opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -342,7 +342,9 @@ BlackoilPropsAdFromDeck::BlackoilPropsAdFromDeck(const BlackoilPropsAdFromDeck&
             TEval.setValue(T.value()[i]);
 
             if (cond[i].hasFreeGas()) {
-                muEval = FluidSystem::oilPvt().saturatedViscosity(pvtRegionIdx, TEval, pEval);
+                const Eval& RsSatEval =
+                    FluidSystem::oilPvt().saturatedGasDissolutionFactor(pvtRegionIdx, TEval, pEval);
+                muEval = FluidSystem::oilPvt().viscosity(pvtRegionIdx, TEval, pEval, RsSatEval);
```

Comment:
I am not sure what are the original "low level" functions mean here. 
It looks like all the `saturated..` functions are split to two steps. In my opinion, it can be easier to use if the two steps can be put in one function to provide a `saturated..` function. 
I am not sure if the original `saturated..` functions are implemented this way though.

Reply:
the thing is that the `saturated*` methods are only called from a very limited number of locations (and in some of them this was either wrong or did not make sense). on the other hand there are about a dozen low-level PVT classes, and each redundant methods increases the difficulty of refactoring these.